# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerPrivate.cpp
# Commit: c0cb3c17f246
# Full Hash: c0cb3c17f2461a19203f3b295a6c56b417483b2a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 84162d09eef8 -r c0cb3c17f246 dom/serviceworkers/ServiceWorkerPrivate.cpp
--- a/dom/serviceworkers/ServiceWorkerPrivate.cpp	Thu Jan 12 15:09:45 2023 +0000
+++ b/dom/serviceworkers/ServiceWorkerPrivate.cpp	Thu Jan 12 15:09:45 2023 +0000
@@ -1566,7 +1566,8 @@
     MOZ_ALWAYS_SUCCEEDS(
         aChannel->GetChannel(getter_AddRefs(underlyingChannel)));
     RefPtr<FetchService> fetchService = FetchService::GetInstance();
-    return fetchService->Fetch(std::move(preloadRequest), underlyingChannel);
+    return fetchService->Fetch(AsVariant(FetchService::NavigationPreloadArgs{
+        std::move(preloadRequest), underlyingChannel}));
   }
   return FetchService::NetworkErrorResponse(NS_ERROR_UNEXPECTED);
 }
